app/database/operator/server_data_operator/transformers/channel.ts

```diff
@@ -42,20 +42,24 @@ export const transformChannelRecord = ({action, database, value}: TransformerArg
 
         // for DM & GM's  channels do not override the display name
         // until we get the new info if there is any
-        let displayName;
-        if (raw.type === General.DM_CHANNEL && (record?.displayName || raw.display_name)) {
-            displayName = raw.display_name || record?.displayName;
-        } else if (raw.type === General.GM_CHANNEL) {
-            const rawMembers = raw.display_name.split(',').length;
-            const recordMembers = record?.displayName.split(',').length || rawMembers;
-
-            if (recordMembers < rawMembers && record.displayName) {
-                displayName = record.displayName;
-            } else {
-                displayName = raw.display_name;
+        let displayName = '';
+        switch (raw.type) {
+            case General.DM_CHANNEL:
+                displayName = raw.display_name || record?.displayName;
+                break;
+            case General.GM_CHANNEL: {
+                const rawMembers = raw.display_name.split(',').length;
+                const recordMembers = record?.displayName.split(',').length || rawMembers;
+                if (recordMembers < rawMembers && record.displayName) {
+                    displayName = record.displayName;
+                } else {
+                    displayName = raw.display_name;
+                }
```

Comment:
Just curiosity, what is this piece of code actually doing?

Reply:
the display name for GM is for example `user1,user2,user3,myuser,user4` and the display name shown in the UI is `User 1, User 2, User 3, User 4` notice how the `myuser` is not included.

So this code is looking at the "format" of the GM and try and determine if we already made the conversion to the way is displayed in the UI and use that value